app/frontend/wallet/shelley/helpers/addresses.ts

```diff
@@ -32,17 +32,21 @@ export const base58AddressToHex = (address: string): HexString => {
   return parsed.toString('hex')
 }
 
-export const stakingAddressFromXpub = (stakeXpub: Buffer, networkId): string => {
+export const stakingAddressFromXpub = (stakeXpub: Buffer, networkId: number): _Address => {
   const addrBuffer: Buffer = packRewardAddress(xpub2blake2b224Hash(stakeXpub), networkId)
   return bech32.encode('stake', addrBuffer)
 }
 
-export const stakingAddressHexFromXpub = (stakeXpub: Buffer, networkId): HexString => {
+export const stakingAddressHexFromXpub = (stakeXpub: Buffer, networkId: number): HexString => {
   const addrBuffer: Buffer = packRewardAddress(xpub2blake2b224Hash(stakeXpub), networkId)
   return Buffer.from(addrBuffer).toString('hex')
 }
 
-export const baseAddressFromXpub = (spendXpub: Buffer, stakeXpub: Buffer, networkId): string => {
+export const baseAddressFromXpub = (
+  spendXpub: Buffer,
+  stakeXpub: Buffer,
+  networkId: number
```

Comment:
it seems sensible to declare a dedicated type for networkId as well which could serve as a guard against passing there some garbage/mixing parameters (eg. protocolMagic with networkId)

Reply:
Sure, to do that I have to type networks, I ve already started working on it but if you don't mind I ll make it into another PR.